jazz_deployments-event-handler/test/test.js

```diff
@@ -1,9 +1,740 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
 const assert = require('chai').assert;
+const expect = require('chai').expect;
+const index = require('../index');
+const request = require('request');
+const awsContext = require('aws-lambda-mock-context');
+const sinon = require('sinon');
+const logger = require("../components/logger.js");
+const configObj = require("../components/config.js");
+
+var reqStub;
+var event = {
+  "Records": [{
+    "kinesis": {
+      "kinesisSchemaVersion": "1.0",
+      "partitionKey": "VALIDATE_INPUT",
+      "sequenceNumber": "49584481860528260622422554690105735408360629197594427394",
+      "data": "eyJJdGVtIjp7IkVWRU5UX0lEIjp7IlMiOiIwODRmOGMzOC1hMDFiLTRhYzktOTQzZS0zNjVmNWRlOGViZTQifSwiVElNRVNUQU1QIjp7IlMiOiIyMDE4LTA1LTE2VDEyOjEyOjQyOjgyMSJ9LCJSRVFVRVNUX0lEIjp7Ik5VTEwiOnRydWV9LCJFVkVOVF9IQU5ETEVSIjp7IlMiOiJKRU5LSU5TIn0sIkVWRU5UX05BTUUiOnsiUyI6IkNSRUFURV9ERVBMT1lNRU5UIn0sIlNFUlZJQ0VfTkFNRSI6eyJTIjoidGVzdC0wMiJ9LCJTRVJWSUNFX0lEIjp7IlMiOiIwOWVkMzI3OS1jOGI5LWUzNjAtMmE3OC00ZTFlZDA5M2U2YTcifSwiRVZFTlRfU1RBVFVTIjp7IlMiOiJTVEFSVEVEIn0sIkVWRU5UX1RZUEUiOnsiUyI6IlNFUlZJQ0VfREVQTE9ZTUVOVCJ9LCJVU0VSTkFNRSI6eyJTIjoic2VydmVybGVzc0B0LW1vYmlsZS5jb20ifSwiRVZFTlRfVElNRVNUQU1QIjp7IlMiOiIyMDE4LTA1LTE2VDEyOjEyOjQxOjA4MyJ9LCJTRVJWSUNFX0NPTlRFWFQiOnsiUyI6IntcInNlcnZpY2VfdHlwZVwiOlwiYXBpXCIsXCJicmFuY2hcIjpcIlwiLFwicnVudGltZVwiOlwibm9kZWpzXCIsXCJkb21haW5cIjpcImphenp0ZXN0XCIsXCJpYW1fcm9sZVwiOlwiYXJuOmF3czppYW06OjE5MjAwNjE0NTgxMjpyb2xlL2dpdGxhYjE4MDUxNV9sYW1iZGEyX2Jhc2ljX2V4ZWN1dGlvbl8xXCIsXCJlbnZpcm9ubWVudFwiOlwiXCIsXCJyZWdpb25cIjpcInVzLWVhc3QtMVwiLFwibWVzc2FnZVwiOlwiaW5wdXQgdmFsaWRhdGlvbiBzdGFydHNcIixcImNyZWF0ZWRfYnlcIjpcInNlcnZlcmxlc3NAdC1tb2JpbGUuY29tXCJ9In19fQ==",
+      "approximateArrivalTimestamp": 1526472764.125
+    },
+    "eventSource": "aws:kinesis",
+    "eventVersion": "1.0",
+    "eventID": "shardId-000000000000:49584481860528260622422554690105735408360629197594427394",
+    "eventName": "aws:kinesis:record",
+    "invokeIdentityArn": "arn:aws:iam::12345678:role/gitlabtest10001_test01",
+    "awsRegion": "us-east-1",
+    "eventSourceARN": "arn:aws:kinesis:us-east-1:100000002:teststream/gitlab10001-events-hub-prod"
+  }]
+};
+var context = {
+  "callbackWaitsForEmptyEventLoop": true,
+  "logGroupName": "/aws/lambda/gitlab1234567-test-name",
+  "logStreamName": "2018/05/16/temp_test0000012910",
+  "functionName": "gitlab001-test-services-handler-prod",
+  "memoryLimitInMB": "256",
+  "functionVersion": "$LATEST",
+  "invokeid": "00001-test-000001",
+  "awsRequestId": "00001-test-000001",
+  "invokedFunctionArn": "arn:aws:lambda:us-east-1:100000001:function:gitlab001-test-services-handler-prod"
+};
+var configData = configObj(context);
```

Comment:
Unit tests missing for failure scenarios. For example, I don't see any of the unit tests exercise the code path where `handleFailedEvents` will be called.

Reply:
Need clarification - handleFailedEvents has only 5 lines of code and is intended to push the failed event to an array 
didn't find a condition to be tested

Reply:
Setup a scenario where you multiple events to process. Stub `processRequest` where you make this `reject` once and `resolve` all the other times. have multiple eve and make one of them.

Reply:
Added - line 184